dialects/affine/passes/lower_for.h

```diff
@@ -6,6 +6,25 @@
 namespace thorin::affine {
 
 /// Lowers the for axiom to actual control flow in CPS.
+/// It basically mimics this implementation:
+/// ```
+/// .con %affine.For_impl
+///     (m: .Nat , n: .Nat , Ts: «n; *»)
+///     (begin: .Idx m, end: .Idx m, step: .Idx m, init: «i: n; Ts#i»,
+///             body: .Cn [iter: .Idx m, acc: «i: n; Ts#i», yield: .Cn [«i: n; Ts#i»]],
+///             exit: .Cn [«i: n; Ts#i»]
+///     ) =
+///     .con head(iter: .Idx m, acc: «i: n; Ts#i») =
+///         .con new_body() = body (iter, acc, .cn acc: «i: n; Ts#i» =
+///             .let `iter = %core.wrap.add %core.mode.nsuw (iter, step);
+///             head (iter, acc));
+///         .con new_exit() = exit (acc);
+///         (new_exit, new_body)#(%core.icmp.ul (iter, end)) ();
+///     head(begin, init);
+/// ```
+/// However, we merge `init`/`acc` into the signature, as it may contain a `mem`.
+/// In this case we have to equip `new_body`/`new_exit` with a `mem` as well.
+/// @todo We probably want to have phases which fix such things so we don't have to do this in C++.
```

Comment:
If I understand the problem correctly, the memory passes like reshape and add_mem should be these passes.

Reply:
I discovered another problem regarding performance. If we have `acc: «1000; T»`, we don't want to merge this with `iter` ...
I'm currently working on things like that, but I'm doing this in a different branch in another PR.

Reply:
The problem are the branches:
```
.con foo(mem: %mem.M, ...) =
    .con t() = load (mem, ...)
    .con f() = store (mem, ...)
    (f, t)#cond ()
```
This is kind of dangerous, because `load`/`store` could now float out of `t`/`f` causing all kind of weird problems. For this reason, my patch in this PR looks for a `mem` and puts it into the `head`er's branch argument. Does AddMem/Reshape cope with that?